go/vt/vttablet/tabletmanager/vreplication/replicator_plan.go

```diff
@@ -646,6 +654,24 @@ func appendFromRow(pq *sqlparser.ParsedQuery, buf *bytes2.Buffer, fields []*quer
 				buf.WriteString(sqltypes.NullStr)
 			} else {
 				vv := sqltypes.MakeTrusted(typ, row.Values[col.offset:col.offset+col.length])
+
+				if conversion, ok := tp.ConvertCharset[col.field.Name]; ok && col.length >= 0 {
+					// Non-null string value, for which we have a charset conversion instruction
+					fromCollation := tp.CollationEnv.DefaultCollationForCharset(conversion.FromCharset)
+					if fromCollation == collations.Unknown {
+						return vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "Character set %s not supported for column %s", conversion.FromCharset, col.field.Name)
+					}
+					toCollation := tp.CollationEnv.DefaultCollationForCharset(conversion.ToCharset)
+					if toCollation == collations.Unknown {
+						return vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "Character set %s not supported for column %s", conversion.ToCharset, col.field.Name)
+					}
+					out, err := charset.Convert(nil, colldata.Lookup(toCollation).Charset(), vv.Raw(), colldata.Lookup(fromCollation).Charset())
+					if err != nil {
+						return sqlerror.NewSQLError(sqlerror.ERTruncatedWrongValueForField, sqlerror.SSUnknownSQLState, "Incorrect string value: %s", err.Error())
+					}
+					vv = sqltypes.MakeTrusted(typ, out)
+				}
```

Comment:
Oh, I want to consolidate this code with the respective code in `bindFieldVal` (just a small refactor). Might do so in this PR, but please review anyway.

Reply:
The consolidation was done.